src/error.rs

```diff
@@ -46,7 +46,7 @@ pub enum Error {
 #[derive(Debug, Clone, PartialEq)]
 pub enum EvalError {
     /// A blame occurred: a contract have been broken somewhere.
-    BlameError(label::Label, CallStack),
+    BlameError(Option<RichTerm>, label::Label, CallStack),
```

Comment:
The structured version is not strictly necessary (we can do that later for all error variant in a pass, as discussed internally), we should at least document what is this `Option<RichTerm>`.

```suggestion
    BlameError {
        /// The argument failing the contract. If the argument has been forced by the contract, `tracked_arg` provides the final value. 
        evaluated_arg: Option<RichTerm>,
        /// The label of the corresponding contract.
        label: label::Label,
        /// The callstack when the blame error was raised.
        call_stack: CallStack,
      },
```